server/db/services/calls.js

```diff
@@ -5,5 +5,20 @@ export default {
     const { campaign_id, contact_id } = params;
 
     return new Call({ campaign_id, contact_id }).save();
+  },
+
+  assignCall: (params) => {
+    const { user_id, campaign_id } = params;
+
+    return new Call()
+      .where({ campaign_id, status: 'AVAILABLE' })
+      .fetch().then((call) => {
```

Comment:
Do you think we should have an error catching statement here and on line 18? I haven't seen a promise in the services I've worked on yet, but I think this could be a place to implement that.